src/general/base_stage/CreatureStageHUDBase.cs

```diff
@@ -781,6 +793,11 @@ protected void UpdateStrain()
         throw new GodotAbstractMethodNotOverriddenException();
     }
 
+    protected virtual bool CanSprint()
```

Comment:
I think that additionally it would be pretty nice if the hotbar button state was set to disabled when not being able to sprint to add a second way to communicate to the player what it means when the strain bar turns red.